odoo/addons/base/models/ir_actions.py

```diff
@@ -196,7 +196,14 @@ def read(self, fields=None, load='_classic_read'):
             for values in result:
                 model = values.get('res_model')
                 if model in self.env:
-                    values['help'] = self.env[model].get_empty_list_help(values.get('help', ""))
+                    rec = self.filtered(lambda r: r.id == values['id'])
+                    eval_ctx = dict(self.env.context)
+                    try:
+                        ctx = safe_eval(rec.context, eval_ctx)
+                    except:
+                        ctx = {}
+
+                    values['help'] = self.with_context(**ctx).env[model].get_empty_list_help(values.get('help', ""))
```

Comment:
What is the purpose of that code exactly ?

Reply:
Yes, we are doing this code for forwarding action context. And this is actual read call for action data from javascript. which later forward this action context with next rpc call to py.

so here `get_empty_list_help` not getting action context resulting into, not getting default_type in crm.lead 

Even, This was disabling feature of displaying alias in help like this( https://drive.google.com/open?id=1mVzTIQjHcVfzjWE7INY1Kx7IoK7WNJzE ).